pageserver/src/object_store.rs

```diff
@@ -14,6 +14,27 @@ pub struct ObjectKey {
     pub buf_tag: BufferTag,
 }
 
+/// A single version of an Object.
+pub struct ObjectVersion {
+    pub lsn: Lsn,
+    pub value: Box<[u8]>,
+}
+
+/// An iterator over `ObjectVersion`
+pub type ObjectsIter<'a> = dyn Iterator<Item = ObjectVersion> + 'a;
+
+/// A single version of an Object, and its tag.
+///
+/// This is basically a flattened form of `(BufferTag, ObjectVersion)`.
+pub struct ObjectAny {
```

Comment:
Maybe we can signal that this is an object within a timeline: `TimelineObject`/`TimelineObjectVersion`
The iterator could be `TimelineObjects`/`TimelineObjectVersions`/`...Iter`

Reply:
I'm not sure I follow you. My first attempt adds 2 types related to `ObjectStore::objects()`: `ObjectAny` is the data, and `AllObjectsIter` is an iterator over that data type.

I definitely believe those need better names, but you just gave me 4 or 5 new names where I expected 2...?

Reply:
Ah these are a few different options for the pair of object + object iterator structs

Format: (object struct name), (object iter struct name)
Proposal 1: TimelineObject, TimelineObjects
Proposal 2: TimelineObjectVersion, TimelineObjectVersions
Proposal 3: TimelineObject, TimelineObjectIter
Proposal 4: TimelineObjectVersion, TimelineObjectVersionIter

Reply:
3 sounds pretty good.